### PR TITLE
feat(api): add admin authorization

### PR DESCRIPTION
**Pull request by namn-grg**
This PR adds admin authorization to the API. The consumer data contains sensitive data like IP address, latitude, and longitude that ideally should not be made publically. Adding authorization allows only the admin to access the sensitive data.

### CODE REVIEW

File: consumer/api.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		if line != "" {
+			apiKeys[line] = true
+		}
+	}
+	return apiKeys, scanner.Err()
+}
+
+func createGetValidatorsHandler(db *sql.DB, apiKeys map[string]bool) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		apiKey := r.Header.Get("API-Key")
```
Comment (mempirate) [inline consumer/api.go]: Let's use the "X-Api-Key" header (which is the standard)

File: consumer/api.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			err := rows.Scan(&vm.PeerID, &vm.ENR, &vm.Multiaddr, &vm.IP, &vm.Port, &vm.LastSeen, &vm.LastEpoch, &vm.ClientVersion, &vm.PossibleValidator, &vm.MaxValidatorCount, &vm.NumObservations, &vm.Hostname, &vm.City, &vm.Region, &vm.Country, &vm.Latitude, &vm.Longitude, &vm.PostalCode, &vm.ASN, &vm.ASNOrganization, &vm.ASNType)
+			if err != nil {
+				http.Error(w, fmt.Sprintf("Error scanning row: %s", err), http.StatusInternalServerError)
+				return
+			}
+
+			// If the user is not an admin, we should not return the sensitive data
+			if !isAdmin {
+				vm.Multiaddr = ""
+				vm.IP = ""
```
Comment (mempirate) [inline consumer/api.go:84]: Let's add ENR to this (ENR also contains the IP address)

File: consumer/consumer.go [hunk 3]
```diff
@@ -49,6 +49,13 @@ func RunConsumer(cfg *ConsumerConfig) {
 	// Set up logging
 	log := log.NewLogger("consumer")
 
+	// Read the API keys
+	apiKeys, err := LoadAPIKeys("api_keys.txt")
```
Comment (mempirate) [inline consumer/consumer.go]: We should read the keys every time there's a new request, not just at startup (so we can update the keys without needing to restart the API)

**Review by mempirate — CHANGES_REQUESTED** [review]
LGTM, some changes requested

File: consumer/consumer.go (hunk 3 above)
Comment (namn-grg) [reply consumer/consumer.go]: done

File: consumer/api.go (hunk 2 above)
Comment (namn-grg) [reply consumer/api.go:84]: ah my bad

File: consumer/api.go (hunk 1 above)
Comment (namn-grg) [reply consumer/api.go]: Right!

**Review by mempirate — APPROVED** [review]

**Pull request merged**